apps/meteor/client/views/admin/import/PrepareImportPage.js

```diff
@@ -40,13 +41,31 @@ function PrepareImportPage() {
 	const [isPreparing, setPreparing] = useSafely(useState(true));
 	const [progressRate, setProgressRate] = useSafely(useState(null));
 	const [status, setStatus] = useSafely(useState(null));
-	const [messageCount, setMessageCount] = useSafely(useState(0));
 	const [users, setUsers] = useState([]);
 	const [channels, setChannels] = useState([]);
+	const [messages, setMessages] = useState([]);
 	const [isImporting, setImporting] = useSafely(useState(false));
 
-	const usersCount = useMemo(() => users.filter(({ do_import }) => do_import).length, [users]);
-	const channelsCount = useMemo(() => channels.filter(({ do_import }) => do_import).length, [channels]);
+	const [usersCount, setUsersCount] = useState(0);
+	const [channelsCount, setChannelsCount] = useState(0);
+
+	// usersCount = selectedUsers.length
+	// channelsCount = selectedChannels.length
+	const selectedUsers = useMemo(() => {
+		const filtered = users.filter(({ do_import }) => do_import);
+		setUsersCount(filtered.length);
+		return filtered;
+	}, [users]);
+
+	const selectedChannels = useMemo(() => {
+		const filtered = channels.filter(({ do_import }) => do_import);
+		setChannelsCount(filtered.length);
+		return filtered;
+	}, [channels]);
```

Comment:
makes no sense creating a memo to set a state inside, if you are memoizing the array you can use array.length with no fear

Reply:
Good point! Thanks for the feedback!!